src/index.ts

```diff
@@ -42,8 +48,18 @@ export function useAsyncReducerState
             >}
     ]   
 {
-    
-    const [state, setState] = useState(initialState);
+    useEffect(() => {
+        if (isInitialStatePromise(initialState)){
+        initialState.then((response: any) => {
```

Comment:
there's some indention issues around here

Reply:
Yes I forgot to format before pushing this commit ( I should enable format on save ). Fixed now